### PR TITLE
feat(docker): update completion (b7b923d, 2022-03-29)

### LINKED ISSUE #10825: Update docker completion script to match upstream
### If the feature request is for a plugin or theme, specify it here.

docker plugin

### If the feature solves a problem you have, specify it here.

Completion script is outdated

### Describe the proposed feature.

Update the completion script to the latest commit from upstream

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by obvionaoe**
Source: https://github.com/docker/cli/blob/b7b923d/contrib/completion/zsh/_docker

Closes #10825

## Changes:

Updates the completion script to the latest from upstream

### CODE REVIEW

**Review by carlosala — APPROVED** [review]

File: plugins/docker/_docker
Comment (mcornella) [inline plugins/docker/_docker:2656]: This modification was to remove `*` from the subcommands:

### with
image

### without
image

I fixed this in https://github.com/ohmyzsh/ohmyzsh/commit/c21ff38b8fda0584208ffd697dac0502e2327466 but forgot to send this upstream.

There's https://github.com/docker/cli/pull/2903 still to be merged, so the Zsh completion is probably not prioritary. Still, I'd like to wait until this patch is merged upstream so as to not introduce a regression.

Comment (mcornella) [reply plugins/docker/_docker:2656]: I've submitted the PR for this particular fix at https://github.com/docker/cli/pull/3540. You can subscribe to that to know when it's merged.

Comment (jmeridth) [reply plugins/docker/_docker:2656]: It merged. So this is good to proceed?

Comment (obvionaoe) [reply plugins/docker/_docker:2656]: Hopefully! @mcornella what do you think?

**Pull request closed without merging**
